### PR TITLE
Avoid panics on login

### PR DESCRIPTION
**Pull request by pmatseykanets**
## Problem

Log messages try to use a variable that may not be set if `EnsureUser()` call fails.
 
## Solution

Use a variable that is set ahead of time instead.
 
### Automated Testing

* Test types added/modified:

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Comment by samjustus** [comment]
#46894

**Review by JonCrowther — APPROVED** [review]

**Review by raulcabello — APPROVED** [review]

File: pkg/auth/providers/publicapi/login.go [hunk 1]
```diff
@@ -237,14 +237,14 @@ func (h *loginHandler) createLoginToken(request *types.APIContext) (v3.Token, st
 		userExtraInfo := providers.GetUserExtraAttributes(providerName, userPrincipal)
 		err = h.tokenMGR.UserAttributeCreateOrUpdate(currUser.Name, userPrincipal.Provider, groupPrincipals, userExtraInfo, loginTime)
 		if err != nil {
-			logrus.Warnf("Error creating or updating userAttribute for %s, retrying: %v", currUser.Name, err)
+			logrus.Warnf("Error creating or updating userAttribute for %s, retrying: %v", userPrincipal.Name, err)
```
Comment (andreas-kupries) [inline pkg/auth/providers/publicapi/login.go:240]: What about `currUser` in `UserAttributeCreateOrUpdate(currUser.Name...` just above the condition ?

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: pkg/auth/providers/publicapi/login.go (hunk 1 above)
Comment (pmatseykanets) [reply pkg/auth/providers/publicapi/login.go:240]: It's unaffected. This call is unreachable if `EnsureUser()` returns an error.

Comment (andreas-kupries) [reply pkg/auth/providers/publicapi/login.go:240]: ok

Comment (pmatseykanets) [reply pkg/auth/providers/publicapi/login.go:240]: The sources of potential panics are L227 and L247

**Pull request merged**